qubes/api/admin.py

```diff
@@ -486,36 +486,6 @@ def vm_volume_resize(self, untrusted_payload):
         finally:  # even if calling qubes.ResizeDisk inside the VM failed
             self.app.save()
 
-    @qubes.api.method('admin.vm.volume.Import', no_payload=True,
-        scope='local', write=True)
-    @asyncio.coroutine
-    def vm_volume_import(self):
-        """Import volume data.
-
-        Note that this function only returns a path to where data should be
-        written, actual importing is done by a script in /etc/qubes-rpc
-        When the script finish importing, it will trigger
-        internal.vm.volume.ImportEnd (with either b'ok' or b'fail' as a
-        payload) and response from that call will be actually send to the
-        caller.
-        """
-        self.enforce(self.arg in self.dest.volumes.keys())
-
-        self.fire_event_for_permission()
```

Comment:
This needs to be added in the new version too, so core-admin extension could deny import based on other factors. It may be need to open-code this function, to hide `internal.` vs `admin.` difference.

Reply:
You mean, fire an event `admin-permission:admin.vm.volume.Import` or `admin-permission:admin.vm.volume.ImportWithSize`, depending on how we are called?

Reply:
Yes.